src/parse/string/shift/operand/enclosed.ts

```diff
@@ -29,44 +41,77 @@ export const parseEnclosed = (s: DynamicState, enclosing: EnclosingChar) => {
         // flags are not currently supported for embedded regex literals
         s.root = typeNode({ basis: "string", regex: `/${token}/` })
     } else {
-        s.root = typeNode({ basis: ["===", token] })
+        const value = hasDateEnclosing(enclosing)
+            ? tryParseDate(
+                  `${enclosing}${token}${openToCloseEnclosingChar[enclosing]}`,
+                  writeInvalidDateMessage(token)
+              )
+            : token
+        s.root = typeNode({ basis: ["===", value] })
     }
 }
 
+type getClosingEnclosed<enclosing extends StartEnclosingChar> =
+    enclosing extends `d${infer quote extends QuoteEnclosingChar}`
+        ? quote
+        : enclosing
+
 export type parseEnclosed<
     s extends StaticState,
-    enclosing extends EnclosingChar,
+    enclosing extends StartEnclosingChar,
     unscanned extends string
-> = Scanner.shiftUntil<unscanned, enclosing> extends Scanner.shiftResult<
-    infer scanned,
-    infer nextUnscanned
->
+> = Scanner.shiftUntil<
+    unscanned,
+    getClosingEnclosed<enclosing>
+> extends Scanner.shiftResult<infer scanned, infer nextUnscanned>
     ? nextUnscanned extends ""
         ? state.error<writeUnterminatedEnclosedMessage<scanned, enclosing>>
         : state.setRoot<
               s,
-              `${enclosing}${scanned}${enclosing}`,
+              `${enclosing}${scanned}${getClosingEnclosed<enclosing>}`,
               nextUnscanned extends Scanner.shift<string, infer unscanned>
                   ? unscanned
                   : ""
           >
     : never
 
-export const enclosingChar = {
+export const quoteEnclosingChar = {
```

Comment:
Can we just git rid of these now that we have the map? It feels a bit like overkill, we should just be able to check these conditions directly.